bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/reconnect/PeriodicReconnectStrategy.java

```diff
@@ -66,15 +78,15 @@ public void lostConnection() {
     }
 
     @Override
-    public void connectionEstablished() {
+    public synchronized void connectionEstablished() {
         if (scheduledTask != null) {
             scheduledTask.cancel(true);
             scheduledTask = null;
         }
```

Comment:
I'm afraid we need to synchronize on a same locker object for instance `scheduler` for `connectionEstablished` and `isReconnecting` and `lostconnection` and `close`. If we just synchronize on the methods, nothing is really gained.

Reply:
Can you please elaborate this a little bit?

* synchronized on non-static member functions should use locker object `this`
* synchronized on static member functions should use the respective `Class` locker object 

Or my knowledge of synchronization is completely wrong. :fearful: 
I will check the docu.

Reply:
http://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.4.3.6

Reply:
Oh pardon me, I have a c++ background and actually learned something, I thought synchronized just make sure that a method is re-entrance thread-safe. Never mind.

Reply:
So, do you approve the changes (using Github's review approval feature) or is anything left? :wink: